engine/storage/configdrive/src/main/java/org/apache/cloudstack/storage/configdrive/ConfigDriveBuilder.java

```diff
@@ -250,6 +298,124 @@ static JsonObject createJsonObjectWithVmData(List<String[]> vmData, String tempD
         return metaData;
     }
 
+    /**
+     * Creates the {@link JsonObject} using @param nic's metadata. We expect the JSONObject to have the following entries:
+     * <ul>
+     *     <li> links </li>
+     *     <li> networks </li>
+     *     <li> services </li>
+     * </ul>
+     */
+    static JsonObject getNetworkDataJsonObjectForNic(NicProfile nic, List<Network.Service> supportedServices) {
+        JsonObject networkData = new JsonObject();
+
+        if (supportedServices.contains(Network.Service.Dhcp)) {
```

Comment:
In case of multiple networks, we will need to pass the configuration of other networks (non-configdrive) here. Otherwise nic won't be setup on the VM.